src/rlookup.c

```diff
@@ -158,7 +160,13 @@ static RLookupKey *RLookup_GetKey_common(RLookup *lookup, const char *name, size
         return NULL;
       }
     } else {
-      key->path = field_name;
+      if (!(flags & RLOOKUP_F_NAMEALLOC)) {
+        key->path = field_name;
+      } else if (name != field_name) {
+        key->path = rm_strndup(field_name, strlen(field_name));
```

Comment:
Should come together with the (commented out) future changes to `RLookupKey_Cleanup`?